de-lib/src/main/java/org/iplantc/de/apps/integration/client/presenter/AppsEditorPresenterImpl.java

```diff
@@ -821,14 +821,16 @@ void clearRegisteredHandlers(){
     }
 
     private void checkForDeprecatedTools(final List<Tool> tools) {
-        if (tools != null && !tools.isEmpty()) {
-            if (tools.stream().anyMatch(Tool::isDeprecated)) {
-                Scheduler.get().scheduleDeferred(() -> {
-                    IplantInfoBox errorsInfo = new IplantInfoBox(appearance.warning(),
-                                                                 appearance.appUsesDeprecatedTools());
+        if (tools != null && tools.size() > 0) {
+            for (Tool t : tools) {
+                if (t!= null && t.isDeprecated()) {
+                    IplantInfoBox errorsInfo =
+                            new IplantInfoBox(appearance.warning(), appearance.appUsesDeprecatedTools());
                     errorsInfo.setIcon(MessageBox.ICONS.warning());
                     errorsInfo.show();
-                });
+                    return;
+
+                }
             }
         }
     }
```

Comment:
I am still confused why I had to re-write this method. Basically I got NPE when restoring an AppEditor window from session when no tools were selected. Also In prod, AppEditor window will not be restored when opened for a creating a new App and changes have to been made.

Reply:
The only thing I can think of is that the `tools` list isn't empty (`tools.size() > 0` should be the same as `!tools.isEmpty()`), but the list contains a null `tool` somehow.

Maybe when the app is getting saved in the session, it's somehow getting saved with a null `tool`?

Reply:
👍